custom_components/tuya_local/devices/goldair_platinum_tower_fan_with_wifi.yaml

```diff
@@ -0,0 +1,61 @@
+name: Goldair Platinum Tower Fan with Wifi
+products:
+  - id: hgsp7xfajllkqujw
+    name: GOLDAIR PLATINUM TOWER FAN WITH WIFI
+primary_entity:
+  entity: fan
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 2
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: 1
+          value: Normal
+        - dps_val: 2
+          value: Nature
+        - dps_val: 3
+          value: Sleep
+        - dps_val: 4
+          value: Child
+    - id: 3
+      type: integer
+      name: speed
+      range:
+        min: 0
+        max: 3
+      mapping:
+        - scale: 0.03
+    - id: 4
+      type: string
+      name: oscillate
+      mapping:
+        - dps_val: "on"
+          value: true
+        - dps_val: "off"
+          value: false
+    - id: 6
+      type: integer
+      name: timer
+    - id: 19
```

Comment:
I would make this a standalone sensor, as temperature is not part of the fan entity standard API.

Under secondary_entities:
```
  - entity: sensor
    class: temperature
    name: Temperature
    dps:
      - id: 19
        type: integer
        name: sensor
        class: measurement
        unit: C
```

Reply:
Good call, I figured there had to be a way to make them their own sensor

Thanks